src/app/api/services/devicesService.ts

```diff
@@ -195,9 +198,15 @@ export const fetchDevices = async (parameters: FetchDevicesParameters): Promise<
         (dataPlaneRequest.headers as any)[HEADERS.CONTINUATION_TOKEN] = parameters.query.continuationTokens[parameters.query.currentPageIndex]; // tslint:disable-line: no-any
     }
 
-    const response = await request(DATAPLANE_CONTROLLER_ENDPOINT, dataPlaneRequest);
-    const result = await dataPlaneResponseHelper(response);
-    return result;
+    try {
+        const response = await request(DATAPLANE_CONTROLLER_ENDPOINT, dataPlaneRequest);
+        AppInsightsClient.getInstance()?.trackEvent({name: TELEMETRY_EVENTS.FETCH_DEVICES}, {status: response.status.toString(), statusText: response.statusText});
+        const result = await dataPlaneResponseHelper(response);
+        return result;
+    } catch (e) {
+        AppInsightsClient.getInstance()?.trackEvent({name: TELEMETRY_EVENTS.FETCH_DEVICES}, {status: 'N/A', statusText: e.toString()});
+        throw (e);
+    }
```

Comment:
Seems like this method has use outside of just fetching devices (even though that's the only place it's used for now).
Possibly in a future PR we could abstract this to a helper function that takes an endpoint, request, and telemetry type;
`export const fetchWithTelemetry(endpoint: string, request: any, telemetryType: {enum_or_telemetry_type_TBD}`